src/native/libs/System.Security.Cryptography.Native/pal_ssl.c

```diff
@@ -1188,6 +1201,89 @@ static int MakeSelfSignedCertificate(X509* cert, EVP_PKEY* evp)
     return ret;
 }
 
+int32_t CryptoNative_GetDefaultSignatureAlgorithms(uint16_t* buffer, int32_t* count)
```

Comment:
do we do similar logic elsewhere? perhaps it's possible to incorporate this there to not do multiple handshakes on init

Reply:
This is running lazily only when the user ever disables any of the sigalgs. There is additional cost for most of the applications.